### PR TITLE
feat(chat): support template markup for Replay functionality (Issue #1592, #1783, #1784, #1788)

### PR DESCRIPTION
**Pull request by IlyaBondar**
**Description:**

support template markup for Replay functionality

Issues:

- Issue #1592 
- Issue #1783 
- Issue #1784 
- Issue #1788

**UI changes**

<Please, provide Screenshots or Figma links>

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1791.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1791.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/ChatInput/ChatInputMessage.tsx [hunk 1]
```diff
@@ -475,6 +494,7 @@ export const ChatInputMessage = ({
             onClose={() => setIsModalVisible(false)}
           />
         )}
+        <ReplyVariables />
```
Comment (denys-kolomiitsev) [inline apps/chat/src/components/Chat/ChatInput/ChatInputMessage.tsx]: I think typo is here
```suggestion
        <ReplayVariables />
```

File: apps/chat/src/components/Chat/ChatInput/PromptVariablesDialog.tsx [hunk 2]
```diff
@@ -84,18 +85,31 @@ export const PromptVariablesDialog: FC<Props> = ({
       );
 
       onSubmit(newContent);
-      onClose();
+      onClose && onClose();
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Chat/ChatInput/PromptVariablesDialog.tsx]:
```suggestion
      onClose?.();
```

File: apps/chat/src/components/Chat/ChatInput/PromptVariablesDialog.tsx [hunk 3]
```diff
@@ -84,18 +85,31 @@ export const PromptVariablesDialog: FC<Props> = ({
...
+  );
+
   const handleKeyDown = useCallback(
     (e: KeyboardEvent<HTMLDivElement>) => {
       if (e.key === 'Enter' && !e.shiftKey) {
         e.preventDefault();
         handleSubmit(e);
       } else if (e.key === 'Escape') {
-        onClose();
+        onClose && onClose();
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Chat/ChatInput/PromptVariablesDialog.tsx]:
```suggestion
        onClose?.();
```

File: apps/chat/src/components/Chat/ChatInput/PromptVariablesDialog.tsx [hunk 4]
```diff
@@ -104,7 +118,7 @@ export const PromptVariablesDialog: FC<Props> = ({
   useEffect(() => {
     const handleOutsideClick = (e: MouseEvent) => {
       if (modalRef.current && !modalRef.current.contains(e.target as Node)) {
-        onClose();
+        onClose && onClose();
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Chat/ChatInput/PromptVariablesDialog.tsx]:
```suggestion
        onClose?.();
```

File: apps/chat/src/components/Chat/ChatInput/PromptVariablesDialog.tsx [hunk 5]
```diff
@@ -84,18 +85,31 @@ export const PromptVariablesDialog: FC<Props> = ({
...
     [onClose, onSubmit, prompt.content, updatedVariables],
   );
 
+  const handleOnBlur = useCallback(
+    (index: number, e: FocusEvent<HTMLTextAreaElement>) => {
+      e.target.value = e.target.value.trim();
+      setUpdatedVariables((prev) => {
+        const updated = [...prev];
+        updated[index].value = e.target.value;
+        return [...updated];
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Chat/ChatInput/PromptVariablesDialog.tsx]:
```suggestion
        updated[index].value = e.target.value;
        return updated;
```

we do not need to spread array which is not part of state

File: apps/chat/src/components/Chat/ChatInput/ReplyVariables.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+import {
+  ConversationsActions,
+  ConversationsSelectors,
+} from '@/src/store/conversations/conversations.reducers';
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
+
+import { PromptVariablesDialog } from './PromptVariablesDialog';
+
+export const ReplyVariables = () => {
```
Comment (denys-kolomiitsev) [inline apps/chat/src/components/Chat/ChatInput/ReplyVariables.tsx]: I think typo is here
```suggestion
export const ReplayVariables = () => {
```

File: apps/chat/src/components/Chat/ChatInput/ReplyVariables.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    ConversationsSelectors.selectIsReplaySelectedConversations,
+  );
+
+  const isReplayRequiresVariables = useAppSelector(
+    ConversationsSelectors.selectIsReplayRequiresVariables,
+  );
+
+  if (!isReplay || !isReplayRequiresVariables) return null;
+
+  return <ReplyVariablesDialog />;
```
Comment (denys-kolomiitsev) [inline apps/chat/src/components/Chat/ChatInput/ReplyVariables.tsx]: I think typo
```suggestion
  return <ReplayVariablesDialog />;
```

File: apps/chat/src/components/Chat/ChatInput/ReplyVariables.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+};
+
+export const ReplyVariablesDialog = () => {
```
Comment (denys-kolomiitsev) [inline apps/chat/src/components/Chat/ChatInput/ReplyVariables.tsx]: Typo also
```suggestion
export const ReplayVariablesDialog = () => {
```

File: apps/chat/src/components/Chat/ChatInput/ReplyVariables.tsx [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    !activeMessage ||
+    !activeMessage.templateMapping ||
+    !Object.keys(activeMessage.templateMapping).length
+  )
+    return null;
+
+  let template = activeMessage.content;
+  Object.entries(activeMessage.templateMapping).forEach(([key, value]) => {
+    template = template.replaceAll(key, value);
+  });
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Chat/ChatInput/ReplyVariables.tsx]:
```suggestion
  const template = Object.entries(activeMessage.templateMapping)
  .reduce((acc, [key, value]) => acc.replaceAll(key, value), activeMessage.content);
```

**Review by denys-kolomiitsev — DISMISSED** [review]

**Review by Alexander-Kezik — APPROVED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1791.nightly-test.deltixhub.io
>E2E tests status: success

**Review by denys-kolomiitsev — APPROVED** [review]

**Pull request merged**